lib/vdc/UnstructuredGrid2D.cpp

```diff
@@ -251,7 +251,18 @@ float UnstructuredGrid2D::GetValueLinear(const CoordType &coords) const
     VAssert(face < GetCellDimensions()[0]);
 
     double value = 0;
-    for (int i = 0; i < nodes.size(); i++) { value += AccessIJK(nodes[i], 0, 0) * lambda[i]; }
+    float  mv = GetMissingValue();
+    for (int i = 0; i < nodes.size(); i++) {
+        float v = AccessIJK(nodes[i], 0, 0);
+        if (v == mv) {
+            if (lambda[i] != 0.0)
+                return (mv);
```

Comment:
Need to free up memory block of `lamba`. Better: use a vector for lambda.

Reply:
Thanks. Done. Replaced with smart pointers.